Lidgren.Network/NetPeerConfiguration.cs

```diff
@@ -344,23 +344,21 @@ public IPAddress LocalAddress
 		}
 
         /// <summary>
-        /// Gets or sets a value indicating whether the library should use IPv6 dual stack mode
+        /// Gets or sets a value indicating whether the library should use IPv6 dual stack mode.
+        /// If you enable this you should make sure that the <see cref="LocalAddress"/> is an IPv6 address.
+        /// Cannot be changed once NetPeer is initialized.
         /// </summary>
-        public bool DualStack
-        {
+        public bool DualStack
+        {
             get { return m_dualStack;  }
-            set
-            {
+            set
+            {
                 if (m_isLocked)
                     throw new NetException(c_isLockedMessage);
                 m_dualStack = value;
-                if (m_dualStack && m_localAddress.Equals(IPAddress.Any))

```

Comment:
I removed this because it just feels like a bad idea to sneakily edit the other property. Especially because it doesn't actually *prevent* you from passing an IPv4 address when you enable dual stack (like by mistake due to order).

I implemented an explicit warning in socket binding if dual stack is being used with an IPv4 address. Should be significantly less error prone. The idea here being that you'd need to explicitly enable both `DualStack` *and* set the listen address to IPv6 so you know what you're getting into.